text/0000-deprecate-authors-field.md

```diff
@@ -0,0 +1,128 @@
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This RFC proposes to deprecate the `package.authors` field of `Cargo.toml`.
+This also implies preventing Cargo from auto-filling it, allowing crates to be
+published to crates.io without the field being present, and avoiding displaying
+its contents on the crates.io and docs.rs UI.
+
+# Motivation
+[motivation]: #motivation
+
+The crates.io registry does not allow users to change the contents of already
+published versions: this is highly desirable to ensure working builds don't
+break in the future, but it also has the unfortunate side-effect of preventing
+people from updating the list of crate authors defined in `Cargo.toml`'s
+`package.authors` field.
+
+This is especially problematic when people change their name or want to remove
+their name from the Internet, and the crates.io team doesn't have any way to
+address that at the moment except for deleting the affected crates or versions
+altogether. We don't do that lightly, but there were a few cases where we were
+forced to do so.
+
+The contents of the field also tend to scale poorly as the size of a project
+grows, with projects either making the field useless by just stating "The
+$PROJECT developers" or only naming the original authors without mentioning
+other major contributors.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+crates.io will allow publishing crates without the `package.authors` field, and
+it will stop showing the contents of the field in its UI (the current owners
+will still be shown). docs.rs will also replace that data with the crate
+owners.
+
+`cargo init` will stop pre-populating the field when running the command, and
+it will not include the field at all in the default `Cargo.toml`. Crate authors
+will still be able to manually include the field before publishing if they so
+choose, even though Cargo will warn when trying to publish those crates.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The implementation of this RFC spans multiple parts of the Rust project:
+
+## Cargo
+
+Cargo will stop fetching the current user's name and email address when running
+`cargo init`, and it will not include the field in the default template for
+`Cargo.toml`. Cargo will also treat the field as deprecated, eventually
+displaying a deprecation warning when someone tries to publish a crate with the
+field set.
+
+## crates.io
+
+crates.io will allow publishing versions without the field and with the field
+empty. The Web UI will remove the authors section, while retaining the current
+owners section.
+
+The API will continue returning the `authors` field in every endpoint which
+currently includes it, but the field will always be empty (even if the crate
+author manually adds data to it). The database dumps will also stop including
+the field.
```

Comment:
This seems to cause a superset of the problems caused by redacting authors in existing versions upon author's explicit request. Are you sure this is justified?

Reply:
As far as I'm aware there is no documented API endpoint that exposes the authorship information, and the database dumps are clearly marked as "experimental". Removing the information from there will mean we can delete it from the crates.io database.